Source/Scripts/Syntax.cs-schema

```diff
@@ -270,11 +324,72 @@ class ExistsExpr : QuantifierExpr
 {
 }
 
+class MapComprehension : ComprehensionExpr
+{
+    Boolean finite;
+    Expression? termLeft;
+}
+
+class LambdaExpr : ComprehensionExpr
+{
+    Specification<FrameExpression> reads;
+}
+
+class FrameExpression : NodeWithOrigin
+{
+    Expression originalExpression;
+    String? fieldName;
+}
+
+class SeqUpdateExpr : Expression
+{
+    Expression seq;
+    Expression index;
+    Expression value;
+}
+
+class SeqConstructionExpr : Expression
+{
+    Type? explicitElementType;
+    Expression n;
+    Expression initializer;
+}
+
+class MultiSetFormingExpr : Expression
+{
+    Expression e;
+}
+
+class MapDisplayExpr : Expression
+{
+    Boolean finite;
+    List<ExpressionPair> elements;
+}
+
+class ExpressionPair
```

Comment:
Consider refactoring the code to avoid having this ugly type here.

Reply:
What would you suggest? The alternatives that come to my mind for the type of `MapDisplayExpr.Elements` are `List<(Expression, Expression)>`, and `List<Expression>` with the invariant that the length is even. I prefer `ExpressionPair` over `(Expression, Expression)` for its explicitness and uniformity; as it stands, schema users don't need to implement special logic for handling tuples (especially in languages without arbitrary-arity and/or anonymous tuples). And in the same vein I think `List<Expression>` with an even-length invariant (or a pair of such lists with an invariant on their lengths being equal) is unfavorable because the invariant isn't encoded in the schema.

Reply:
How about making `ExpressionPair` a nested type of `MapDisplayExpr` and naming it `Entry` ?

Reply:
Based on discussion offline, I've renamed `ExpressionPair` to `MapDisplayEntry`. But I decided not to move it into `MapDisplayExpr` because doing so would require making the schema- and deserializer-generation logic able to handle nested class names, which I didn't feel was worthwhile at this point. Please let me know if you feel otherwise.